frontend/src/components/MeetingRoom/LayoutButton/LayoutButton.spec.tsx

```diff
@@ -1,30 +1,61 @@
 import { render, screen } from '@testing-library/react';
 import { describe, expect, it, vi, Mock, beforeEach } from 'vitest';
 import userEvent from '@testing-library/user-event';
+import { SessionContextType } from '../../../Context/SessionProvider/session';
 import LayoutButton from './LayoutButton';
 import useSessionContext from '../../../hooks/useSessionContext';
 
 vi.mock('../../../hooks/useSessionContext');
+const mockUseSessionContext = useSessionContext as Mock<[], SessionContextType>;
+const sessionContext = {
+  setLayoutMode: vi.fn(),
+} as unknown as SessionContextType;
 
 describe('LayoutButton', () => {
   beforeEach(() => {
     vi.clearAllMocks();
+    mockUseSessionContext.mockReturnValue(sessionContext);
   });
   it('should render the sidebar view icon if it is an active speaker layout', () => {
-    (useSessionContext as Mock).mockReturnValue({ layoutMode: 'active-speaker' });
-    render(<LayoutButton isScreenSharePresent={false} />);
+    sessionContext.layoutMode = 'active-speaker';
+    const { rerender } = render(
+      <LayoutButton isScreenSharePresent={false} isPinningPresent={false} />
+    );
+    rerender(<LayoutButton isScreenSharePresent={false} isPinningPresent={false} />);
     expect(screen.getByTestId('ViewSidebarIcon')).toBeInTheDocument();
   });
 
+  it('should call the set layout mode function when triggered', async () => {
+    sessionContext.layoutMode = 'active-speaker';
+    const { rerender } = render(
+      <LayoutButton isScreenSharePresent={false} isPinningPresent={false} />
+    );
+    rerender(<LayoutButton isScreenSharePresent={false} isPinningPresent={false} />);
+    const button = await screen.findByRole('button');
+    await userEvent.click(button);
+    expect(sessionContext.setLayoutMode).toHaveBeenCalled();
+
+    sessionContext.layoutMode = 'grid';
+    rerender(<LayoutButton isScreenSharePresent={false} isPinningPresent={false} />);
+    await userEvent.click(button);
+    expect(sessionContext.setLayoutMode).toHaveBeenCalled();
+  });
```

Comment:
I added this unit test for the feedback here: https://github.com/Vonage/vonage-video-react-app/pull/115/files#r1984103601

I settled for doing `expect(sessionContext.setLayoutMode).toHaveBeenCalled();` without `toHaveBeenCalledWith` since `setLayoutMode` has a type `Dispatch<SetStateAction<LayoutMode>>` and afaik there aren't any good ways to unit test this kind of objects